include/swift/Remote/MetadataReader.h

```diff
@@ -166,6 +181,51 @@ class TypeDecoder {
       return Builder.createFunctionType(arguments, argsAreInOut,
                                         result, flags);
     }
+    case NodeKind::ImplFunctionType: {
+      // Minimal support for lowered function types. These come up in
+      // reflection as capture types. For the reflection library's
+      // purposes, the only part that matters is the convention.
+      FunctionTypeFlags flags;
+
+      for (unsigned i = 0; i < Node->getNumChildren(); i++) {
+        auto child = Node->getChild(i);
+
+        if (child->getKind() == NodeKind::ImplConvention) {
+          if (!child->hasText())
+            return BuiltType();
+
+          auto &text = child->getText();
+
+          if (text == "@convention(thin)") {
+            flags =
+              flags.withConvention(FunctionMetadataConvention::Thin);
+          }
+        } else if (child->getKind() == NodeKind::ImplFunctionAttribute) {
+          if (!child->hasText())
+            return BuiltType();
+
+          auto &text = child->getText();
+          if (text == "@convention(c)") {
+            flags =
+              flags.withConvention(FunctionMetadataConvention::CFunctionPointer);
+          } else if (text == "@convention(block)") {
+            flags =
+              flags.withConvention(FunctionMetadataConvention::Block);
+          }
+        }
+      }
+
+      // Completely punt on argument types and results.
```

Comment:
I don't think we care about it unless at some point we add support for function reflection and dynamic calls...